.github/workflows/CI.yml

```diff
@@ -39,19 +39,13 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
```

Comment:
Question: why do we use the `setup-python` action instead of letting uv manage Python versions here?

Reply:
Cause that does not work well with CodSpeed/Valgrind. Valgrind could be fixed when we run it by specifying `uv run` again in the CI, but we do not have access to CodSpeed's run command, so there's no way to fix that.

Besides, uv is a bit slower because GitHub caches these Python downloads (uv needs to bring them in from outside). I'm sure they could perhaps be cached but I don't really know how, and the CodSpeed/Valgrind thing was already a bit of a dealbreaker.